manager/manager.py

```diff
@@ -0,0 +1,1105 @@
+        """
+        sub_names = self._process_names(sub_names, "sub")
+
+        if make_ses_tree:
+            if ses_names is None:
+                ses_names = [self.cfg["ses_prefix"] + "001"]  # TODO: default ses name
+            else:
+                ses_names = self._process_names(ses_names, "ses")
+        else:
+            ses_names = []
+
+        self._make_directory_trees(
+            data_type, sub_names, ses_names, make_ses_tree, process_names=False
+        )
+
+    def make_ses_folder(
+        self,
+        data_type: str,
+        sub_names: Union[str, list],
+        ses_names: Union[str, list],
+        make_ses_tree: bool = True,
+    ):
+        """
+        See make_sub_folder() for inputs.
+        """
+        self._make_directory_trees(data_type, sub_names, ses_names, make_ses_tree)
+
+    def make_ses_tree(
+        self, data_type: str, sub_names: Union[str, list], ses_names: Union[str, list]
+    ):
+        """
+        See make_sub_folder() for inputs.
+        """
+        self._make_directory_trees(data_type, sub_names, ses_names)
+
+    # --------------------------------------------------------------------------------------------------------------------
+    # Public File Transfer
+    # --------------------------------------------------------------------------------------------------------------------
+
+    def upload_data(
+        self,
+        data_type: str,
+        sub_names: Union[str, list],
+        ses_names: Union[str, list],
+        preview: bool = False,
+    ):
+        """
+        Upload data from a local machine to the remote project folder.
+        In the case that a file / folder exists on the remote and local, the local will
+        not be overwritten even if the remote file is an older version.
+
+        TODO: this is highly sub-optimal performance wise when downloading using the "all" option
+              as opens a connection each time to check what subdirectories are present on the remote.
+              Also, re-opens connection for each ses folder, although this is not be too bad as each ses
+              folder will usually contain quite a lot of data.
+
+        :param data_type: see make_sub_folder()
+        :param sub_names: a list of sub names as accepted in make_sub_folder(). "all" will search for all
+                          sub- folders in the data type folder to upload.
+        :param ses_names: a list of ses names as accepted in make_sub_folder(). "all" will search each
+                          sub- folder for ses- folders and upload all.
+        :param preview: perform a dry-run of upload, to see which files are moved.
+        """
+        self._transfer_sub_ses_data("upload", data_type, sub_names, ses_names, preview)
+
+    def download_data(
+        self, data_type: str, sub_names: Union[str, list], ses_names: Union[str, list]
```

Comment:
`data_type` could be confusing, this could mean, e.g. `pathlib.Path`, `.tiff`, or `ephys`. Maybe something like `experiment_type` would be clearer?